### PR TITLE
Miscellaneous Changes

### PR DESCRIPTION
**Pull request by move47**
- Remove computation of `vid_commitment` which was used earlier just to match corresponding DA and QC Proposals.
- Provides builder fee info requirements mentioned https://www.notion.so/espressosys/Builder-fee-payments-4c27ef6c401544359df3ad8a7c59ef08?pvs=4#a9ae65113c2347c799bca5dc42fd725d

### CODE REVIEW

File: src/builder_state.rs [hunk 1]
```diff
@@ -67,7 +67,7 @@ pub struct QCMessage<TYPES: NodeType> {
 /// Request Message to be put on the request channel
 #[derive(Clone, Debug, PartialEq)]
 pub struct RequestMessage {
-    pub requested_vid_commitment: VidCommitment,
+    pub requested_builder_commitment: BuilderCommitment,
```
Comment (nyospe) [inline src/builder_state.rs:70]: As long as we're consistent, I guess this is fine.

File: src/builder_state.rs [hunk 2]
```diff
@@ -93,6 +93,25 @@ pub enum Status {
     ShouldContinue,
 }
 
+/// Builder State to hold the state of the builder
+#[derive(Debug, Clone)]
+pub struct BuiltFromInfo<TYPES: NodeType> {
```
Comment (nyospe) [inline src/builder_state.rs:98]: Okay... `BuiltFromProposedBlock`, maybe?

File: src/builder_state.rs [hunk 3]
```diff
@@ -276,7 +292,7 @@ impl<TYPES: NodeType> BuilderProgress<TYPES> for BuilderState<TYPES> {
         }
         // Do the validation check
         else if da_msg.proposal.data.view_number.get_u64()
-            != self.built_from_view_vid_leaf.0.get_u64() + 1
+            != self.built_from_info.view_number.get_u64() + 1
         {
             tracing::info!("View number is not equal to built_from_view + 1, so returning");
```
Comment (nyospe) [inline src/builder_state.rs:297]: And maybe update this message to match.

File: src/builder_state.rs [hunk 4]
```diff
@@ -108,19 +127,17 @@ pub struct BuilderState<TYPES: NodeType> {
...
     pub quorum_proposal_payload_commit_to_quorum_proposal:
-        HashMap<VidCommitment, QuorumProposal<TYPES>>,
+        HashMap<BuilderCommitment, QuorumProposal<TYPES>>,
 
-    /// view number of the basis for this block
-    /// vid commitment of the basis for this block
-    /// Commitment<Leaf<TYPES>> => QuorumCertificate::vote_commitment
-    pub built_from_view_vid_leaf: (TYPES::Time, VidCommitment, Commitment<Leaf<TYPES>>),
+    /// the spawned from info for a builder state
+    pub built_from_info: BuiltFromInfo<TYPES>,
```
Comment (nyospe) [inline src/builder_state.rs:140]: And, likewise, this name can be `built_from_proposed_block`

File: src/builder_state.rs [hunk 5]
```diff
@@ -303,15 +320,22 @@ impl<TYPES: NodeType> BuilderProgress<TYPES> for BuilderState<TYPES> {
         // set the total nodes required for the VID computation // later required in the build_block
         self.total_nodes = NonZeroUsize::new(total_nodes).unwrap();
 
-        let payload_vid_commitment = vid_commitment(&encoded_txns, total_nodes);
+        // form a block payload from the encoded transactions
```
Comment (nyospe) [inline src/builder_state.rs:323]: 👍  And that's the crucial part.

**Review by nyospe — APPROVED** [review]
There are a couple of places where I'd prefer a different name, and a couple of comments and log messages that could be cleaned up a little, but nothing that should prevent this going in.

File: src/builder_state.rs (hunk 3 above)
Comment (move47) [reply src/builder_state.rs:298]: https://github.com/EspressoSystems/hotshot-builder-core/pull/77/commits/1b6f986c134cabf2e600325c4a4bb2361fce2a7f

Comment (move47) [reply src/builder_state.rs:298]: I think that message is good to debug. I had to explicitly kept it like that.

File: src/builder_state.rs (hunk 2 above)
Comment (move47) [reply src/builder_state.rs]: https://github.com/EspressoSystems/hotshot-builder-core/pull/77/commits/1b6f986c134cabf2e600325c4a4bb2361fce2a7f

**Review by nyospe — APPROVED** [review]
LGTM. When the CI finishes, go ahead and merge. We can hold off on the tag until either you get the other fixes in or we absolutely need it for sequencer.

**Pull request merged**